data_model/src/lib.rs

```diff
@@ -864,6 +864,83 @@ mod model {
         /// in the next request to continue fetching results of the original query
         pub cursor: crate::query::cursor::ForwardCursor,
     }
+
+    /// String containing serialized valid JSON.
+    ///
+    /// This string is guaranteed to be parsed as JSON.
+    #[derive(Display, Default, Debug, Clone, Eq, Encode, Decode, Ord, PartialOrd, IntoSchema)]
+    #[ffi_type(unsafe {robust})]
+    #[repr(transparent)]
+    #[display(fmt = "{}", "0")]
+    pub struct JsonString(pub(super) String);
+}
+
+impl JsonString {
+    /// Deserialize JSON into something
+    /// # Errors
+    /// See [`serde_json::from_str`].
+    pub fn deserialize<'a, T>(&'a self) -> serde_json::Result<T>
+    where
+        T: Deserialize<'a>,
+    {
+        serde_json::from_str(&self.0)
+    }
+
+    /// Serializes a value into [`JsonString`]
+    /// # Errors
+    /// See [`serde_json::to_string`].
+    pub fn serialize<T: serde::Serialize>(value: &T) -> serde_json::Result<Self> {
+        let serialized = serde_json::to_string(value)?;
+        // the string was obtained from serde_json serialization,
+        // so it should be a valid JSON string
+        Ok(Self(serialized))
+    }
```

Comment:
why do you have these 2 special functions when they are implemented as serde traits?

Reply:
Because these functions explicitly use `serde_json` serialization, and the caller doesn't need to care about it.